internal/yml/merge.go

```diff
@@ -98,6 +105,33 @@ func mergeSubTrees(src, dest *yaml.Node) *yaml.Node {
 	return nil
 }
 
+func findInsertionIndex(src, dest *yaml.Node, key string) int {
```

Comment:
Not that it matters, but that can't be fast...

Reply:
I'm sure you've heard what Confucius had to say about that:
> Premature optimization is the root of all Evil.
That's a `O(n^2)` algorithm, but even a `O(n^3)` or worse can be blazingly fast when `n` is rather small (which will always be the case for Joy releases, even super hefty ones, probably staying in the range of 5 to 50 siblings, tops). Not even considering that merges are performed on user's machine, only a few at a time, with wait time barely perceptible.  I'd say we wait until it becomes an issue (if ever) and then revisit, but I would be very curious and interested to know if you have any alternative algorithm achieving same result more efficiently (even if we don't necessarily implement it right now, just for my bettering! 😁)

Reply:
Yep, hence the "Not that it matters" 🙂